var/spack/repos/builtin/packages/py-pyarrow/package.py

```diff
@@ -62,6 +65,13 @@ class PyPyarrow(PythonPackage, CudaPackage):
 
     patch("for_aarch64.patch", when="@0 target=aarch64:")
 
+    def setup_build_environment(self, env):
```

Comment:
https://github.com/spack/spack/issues/35160#issuecomment-1406463878 for context / info

Reply:
It might be better to decouple this from `install_options`. These flags are only for pip, I'm planning on adding support for build/installer in the future which will likely have different flags.